src/storage/src/source/kafka.rs

```diff
@@ -145,6 +145,14 @@ pub struct KafkaResumeUpperProcessor {
     progress_statistics: Arc<Mutex<PartialProgressStatistics>>,
 }
 
+/// Computes whether this worker is responsible for consuming a partition. It assigns partitions to
+/// workers in a round-robin fashion, starting at an arbitrary worker based on the hash of the
+/// source id.
+fn responsible_for_pid(config: &RawSourceCreationConfig, pid: i32) -> bool {
+    let pid = usize::try_from(pid).expect("positive pid");
+    ((config.responsible_worker(config.id) + pid) % config.worker_count) == config.worker_id
+}
```

Comment:
doesn't this only work in a round-robin fashion if the `pids` are all sequential? Is that something we can assume is guaranteed, or should we instead use the index of the partition among all partitions instead of its id?

Reply:
Yes and yes :) It is guaranteed that partitions are consecutive and this is an assumption that is already baked into how we handle capabilities where we only hold onto the `(max_pid, inf)` range for future partitions and everything below `max_pid` is active.